packages/matchbox/src/components/Modal/Modal.js

```diff
@@ -1,88 +1,144 @@
+            <div ref={el => (content = el)}>
+              <WindowEvent event="keydown" handler={handleKeydown} />
+
+              <WindowEvent event="click" handler={handleOutsideClick} />
 
-    if (open && onClose) {
-      onKey('escape', onClose)(e);
-    }
-  }
-
-  render() {
-    const {
-      onClose,
-      open,
-      children,
-      className,
-      showCloseButton,
-      ...rest
-    } = this.props;
-
-    const modalClasses = classnames(
-      styles.Modal,
-      open && styles.open,
-      className
-    );
-
-    return (
-      <div className={modalClasses} onClose={onClose} {...rest} ref={(node) => this.container = node} role="dialog" aria-modal="true">
-        <Grid center='xs' middle='xs' className={styles.Grid}>
-          <Grid.Column xs={11} md={9} xl={7}>
-            <Content contentRef={(node) => this.content = node} open={open}>
-              <WindowEvent event='keydown' handler={this.handleKeyDown} />
-              <WindowEvent event='click' handler={this.handleOutsideClick} />
               {showCloseButton && (
-                <Button className={styles.CloseButton} flat onClick={onClose} data-id='modal-close'>
-                  Close <Close />
-                </Button>
+                <StyledCloseButton flat onClick={onClose} data-id="modal-close">
+                  <ScreenReaderOnly>Close</ScreenReaderOnly>
+
+                  <Close size={34} />
+                </StyledCloseButton>
               )}
+
               {children}
-            </Content>
-          </Grid.Column>
-        </Grid>
-      </div>
-    );
-  }
+            </div>
+          </ModalContent>
+        </StyledWrapper>
+      </StyledBase>
+    </Portal>
+  );
 }
 
+function ModalContent(props) {
+  const { open, children, maxWidth } = props;
+  const content = useRef(null);
+
+  return (
+    <StyledFocusLock disabled={!open} maxWidth={maxWidth}>
+      <Transition
+        mountOnEnter
+        unmountOnExit
+        in={open}
+        timeout={{
+          enter: 300,
+          exit: 150,
+        }}
+      >
+        {/* Negative `tabIndex` required to programmatically focus */}
+        {state => (
+          <StyledContent state={state} tabIndex="-1" ref={content} data-id="modal-content-wrapper">
+            {children}
+          </StyledContent>
+        )}
+      </Transition>
+    </StyledFocusLock>
+  );
+}
+
+Modal.propTypes = {
+  /**
+   * Controlled open state of the modal
+   */
+  open: PropTypes.bool,
+  /**
+   * An optional function that is called on key down 'Escape' and on click outside modal content
+   */
+  onClose: PropTypes.func,
+  /**
+   * Modal content
+   */
+  children: PropTypes.node,
+  showCloseButton: PropTypes.bool,
+  maxWidth: PropTypes.string,
+  /**
+   * Controls the target container ID for the rendering React portal
+   */
+  containerId: PropTypes.string,
+};
+
+Modal.defaultProps = {
+  maxWidth: '800px',
+  containerId: 'modal-portal',
```

Comment:
Not sure this should have a default, the Portal component defaults to document.body without an id

Reply:
Also Tooltip uses the prop `portalId` if you wanted to align the prop name